### PR TITLE
bugs fixed

### PR DESCRIPTION
**Pull request by mbogor**
#### What's this PR do?
 + fixes a ctlOnline bug
+ fixes a platformId bug
#### Where should the reviewer start?
platform creation and ctlOnline scripts
#### How should this be manually tested?
run the platform creation job
#### What are the relevant tickets?
https://admin.crisistextline.org/jira/browse/INT-513

### CODE REVIEW

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 1]
```diff
@@ -98,10 +100,20 @@ function gradeLogs(courseID){
...
 }
 
+function getPlatformId(data){
+  if (data.platformReadyId) resolve(data);
+  return new Promise((resolve, reject)=>{
+    canvas.getAssignments(data.courseID, 'Platform')
+    .then((response)=>{
+      data.platformReadyId = response[0].id;
+      resolve(data);
+    })
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: ;

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 2]
```diff
@@ -98,10 +100,20 @@ function gradeLogs(courseID){
...
+  })
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: ;

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 3]
```diff
@@ -24,6 +25,7 @@ function graduateCanvasUserToPlatform() {
     courses.forEach(function (course) {
 
       gradeLogs(course.id)
+      .then(getPlatformId)
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: PlatformId meaning the id of the assignment platform ready in canvas? Can you add 'Ready' to the name so it is clearer?

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 4]
```diff
@@ -190,12 +203,10 @@ function updatePlatform(body) {
 }
 
 // HELPER FUNCTIONS
-function ctlOnlineGraduate (email) {
-  return ctlOnline.uidFromEmail(email)
-    .then(ctlOnline.addGraduateRole)
-    .then(() => email)
+function ctlOnlineGraduate (id) {
+    return ctlOnline.addGraduateRole(id)
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: Isn't this their canvas id and not their CTL online Id now?

**Comment by apackin** [comment]
I see how this would fix a bug where the assignment in canvas is not called exactly platformReady: 'Platform Ready!'.

What's the bug in ctlOnline that it fixes? I'm confused as to were the ctlOnline id is coming from now.

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js (hunk 3 above)
Comment (mbogor) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: no

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 5]
```diff
@@ -191,11 +204,11 @@ function updatePlatform(body) {
 
 // HELPER FUNCTIONS
 function ctlOnlineGraduate (email) {
-  return ctlOnline.uidFromEmail(email)
+   return ctlOnline.uidFromEmail(email)
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: Space

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 6]
```diff
@@ -144,7 +157,7 @@ function graduateStudents(grads){
           notifySlack(user.name);
         })
         .then(()=>{
-          ctlOnlineGraduate(user.login_id);
+          ctlOnlineGraduate(user.id);
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: ctlOnlineGraduate(user.id) is their email address?

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 7]
```diff
@@ -131,6 +143,7 @@ function getQuizLogs(potentials){
 }
 
 function graduateStudents(grads){
+
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: space

File: jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js [hunk 8]
```diff
@@ -7,7 +7,7 @@ const internalRequest = require('request');
 const moment = require('moment');
 const throttler = require('throttled-request')(internalRequest);
 const CronJob = require('cron').CronJob;
-const SLACK_CHANNEL = '#graduates';
+const SLACK_CHANNEL = '#testtest';
```
Comment (apackin) [inline jobs/training/pollCanvasForGraduatedUsersThenCreatePlatformAccount.js]: revert?

**Comment by apackin** [comment]
👍

**Pull request merged**